test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -803,19 +806,19 @@ public void testInitMetadataColumns_falseBlank() throws Exception {
   @Test
   public void testInitMetadataColumns_trueBlank() throws Exception {
     executeUpdate(
-        "create table data(id int, foo varchar(20), bar varchar(20))");
-    executeUpdate("insert into data(id, foo, bar) "
+        "create table data(ID int, FOO varchar(20), BAR varchar(20))");
```

Comment:
That combined with the case-sensitivity of keys in Metadata.equals, which is the real issue. Please add a comment about that. Also, if it's just case-preserving, why do you need uppercase everywhere, and which one actually matters (the table definition or the query)?